pkg/virtctl/root.go

```diff
@@ -36,6 +36,29 @@ func NewVirtctlCommand() *cobra.Command {
 			fmt.Fprint(cmd.OutOrStderr(), cmd.UsageString())
 		},
 	}
+	completionCmd := &cobra.Command{
+		Use:   "completion",
+		Short: "Generates completion scripts for the specified shell",
+		Args:  cobra.MaximumNArgs(1),
+		Long: `To load completion run
+. <(virtctl completion (SHELL TYPE))
+
+To configure bash shell to load completions for each session add to bashrc
+
+# ~/.bashrc or ~/.profile
+. <(virtctl completion (SHELL TYPE))
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			switch {
+			case len(args) == 0:
+				rootCmd.GenBashCompletion(os.Stdout)
```

Comment:
I would prefer to align argument handling with kubectl, and return an error if no arg is given, instead of defaulting to bash. Also, please check for `len(args) > 1` for giving a better error message. See https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/completion/completion.go#L118